pkg/pillar/cmd/baseosmgr/handlebaseos.go

```diff
@@ -211,6 +212,24 @@ func doBaseOsStatusUpdate(ctx *baseOsMgrContext, uuidStr string,
 		changed = true
 	}
 
+	// Check to avoid upgrading from not EVE-k to EVE-k since that
```

Comment:
And the opposite way (from k to kvm) is supported?

Reply:
That is an interesting question. @zedi-pramodh @andrewd-zededa is this something we should also check for and block?

Reply:
Why can't you do exactly reverse of what you are doing in this PR ? If base OS HV type is k and new image is kvm fail the upgrade ?

Reply:
Also unfortunately these changes only help in the future, existing eve installs will fail anyway :(

Reply:
Fixed (and added a bonus commit!)